### PR TITLE
[IMP] Added br filtering to override methods in account invoice and a…

### PR DESCRIPTION
**Pull request by ghost**
…ccount invoice line models

### CODE REVIEW

**Comment by ghost** [comment]
@danimaribeiro ,
Can you help me to solve travis errors? I can't find where is the problem.
Thank you so much!!

File: br_nfe/tests/test_inutilizacao.py [hunk 1]
```diff
@@ -176,7 +176,10 @@ def test_inutilizacao_falha_schema(self, inutilizar, validar):
 
     @patch('odoo.addons.br_nfe.models.invoice_eletronic.valida_nfe')
     @patch('odoo.addons.br_nfe.models.inutilized_nfe.inutilizar_nfe')
-    def test_inutilizacao_2_sequences(self, inutilizar, validar):
+    @patch('odoo.addons.br_localization_filtering.models.br_localization_filtering.BrLocalizationFiltering._is_user_br_localization')  # noqa  java feelings
+    def test_inutilizacao_2_sequences(self, inutilizar, validar,
+                                      br_localization):
```
Comment (danimaribeiro) [inline br_nfe/tests/test_inutilizacao.py]: The order here is wrong

File: br_nfe/tests/test_inutilizacao.py [hunk 2]
```diff
@@ -215,7 +218,10 @@ def test_inutilizacao_2_sequences(self, inutilizar, validar):
 
     @patch('odoo.addons.br_nfe.models.invoice_eletronic.valida_nfe')
     @patch('odoo.addons.br_nfe.models.inutilized_nfe.inutilizar_nfe')
-    def test_inutilizacao_return_ok(self, inutilizar, validar):
+    @patch('odoo.addons.br_localization_filtering.models.br_localization_filtering.BrLocalizationFiltering._is_user_br_localization')  # noqa  java feelings
+    def test_inutilizacao_return_ok(self, inutilizar, validar,
+                                    br_localization):
+        br_localization.return_value = True
```
Comment (danimaribeiro) [inline br_nfe/tests/test_inutilizacao.py:224]: Here too:
The order here is wrong

**Comment by danimaribeiro** [comment]
@svazquezco I think the only problem is the order of paremeters in tests for patching methods.

**Comment by svazquezco** [comment]
Thank you @danimaribeiro , I was looking inside of the function and I didn't see the problem was the order of the parameters.

I'll fix on Friday. Tomorrow is holiday here. Have a nice day!

**Comment by ghost** [comment]
@danimaribeiro , please, can you help with Travis error? I've reviewed the methods are called by create_invoice and compute fields but I didn't find the error.

**Comment by danimaribeiro** [comment]
Yes, I'll check it.
But is one tax that is not being computed.

**Comment by ghost** [comment]
Hi @danimaribeiro ,
I've taken long time looking for the mistake, I detected br_localization is not set True in invoice creation so I added the patch in setUp method but didn't solve the problem.
Maybe I'm wrong because I don't have knowledge of Brazilian localization but is it possible test files are wrong?

This is xml created by the first invoice (default_invoice and partner_fisica):

```
<COFINS>
			<COFINSAliq>
				<CST>01</CST>
				<vBC>0.00</vBC>
				<pCOFINS>0.00</pCOFINS>
				<vCOFINS>0.00</vCOFINS>
			</COFINSAliq>
		</COFINS>
		<ICMSUFDest>
			<vBCUFDest>1000.00</vBCUFDest>
			<pFCPUFDest>0.00</pFCPUFDest>
			<pICMSUFDest>17.00</pICMSUFDest>
			<pICMSInter>7.00</pICMSInter>
			<pICMSInterPart>80.00</pICMSInterPart>
			<vFCPUFDest>0.00</vFCPUFDest>
			<vICMSUFDest>80.00</vICMSUFDest>
			<vICMSUFRemet>20.00</vICMSUFRemet>
		</ICMSUFDest>
	</imposto>
</det>
<total>
	<ICMSTot>
		<vBC>0.00</vBC>
		<vICMS>0.00</vICMS>
		<vICMSDeson>0.00</vICMSDeson>
		<vFCPUFDest>0.00</vFCPUFDest>
		<vICMSUFDest>80.00</vICMSUFDest>
		<vICMSUFRemet>20.00</vICMSUFRemet>
		<vFCP>0.00</vFCP>
		<vBCST>0.00</vBCST>
		<vST>0.00</vST>
		<vFCPST>0.00</vFCPST>
		<vFCPSTRet>0.00</vFCPSTRet>
		<vProd>1000.00</vProd>
		<vFrete>0.00</vFrete>
		<vSeg>0.00</vSeg>
		<vDesc>0.00</vDesc>
		<vII>0.00</vII>
		<vIPI>0.00</vIPI>
		<vIPIDevol>0.00</vIPIDevol>
		<vPIS>0.00</vPIS>
		<vCOFINS>0.00</vCOFINS>
		<vOutro>0.00</vOutro>
		<vNF>2000.00</vNF>
		<vTotTrib>0.00</vTotTrib>
	</ICMSTot>

```

And this is the test file:

```
<COFINS>
		<COFINSAliq>
			<CST>01</CST>
			<vBC>0.00</vBC>
			<pCOFINS>0.00</pCOFINS>
			<vCOFINS>0.00</vCOFINS>
		</COFINSAliq>
	</COFINS>
	<ICMSUFDest>
		<vBCUFDest>1000.00</vBCUFDest>
		<pFCPUFDest>0.00</pFCPUFDest>
		<pICMSUFDest>17.00</pICMSUFDest>
		<pICMSInter>7.00</pICMSInter>
		<pICMSInterPart>80.00</pICMSInterPart>
		<vFCPUFDest>0.00</vFCPUFDest>
		<vICMSUFDest>80.00</vICMSUFDest>
		<vICMSUFRemet>20.00</vICMSUFRemet>
	</ICMSUFDest>
	</imposto>
</det>
<total>
	<ICMSTot>
		<vBC>0.00</vBC>
		<vICMS>0.00</vICMS>
		<vICMSDeson>0.00</vICMSDeson>
		<vFCPUFDest>0.00</vFCPUFDest>
		<vICMSUFDest>0.00</vICMSUFDest>
		<vICMSUFRemet>0.00</vICMSUFRemet>
		<vFCP>0.00</vFCP>
		<vBCST>0.00</vBCST>
		<vST>0.00</vST>
		<vFCPST>0.00</vFCPST>
		<vFCPSTRet>0.00</vFCPSTRet>
		<vProd>1000.00</vProd>
		<vFrete>0.00</vFrete>
		<vSeg>0.00</vSeg>
		<vDesc>0.00</vDesc>
		<vII>0.00</vII>
		<vIPI>0.00</vIPI>
		<vIPIDevol>0.00</vIPIDevol>
		<vPIS>0.00</vPIS>
		<vCOFINS>0.00</vCOFINS>
		<vOutro>0.00</vOutro>
		<vNF>0.00</vNF>
		<vTotTrib>0.00</vTotTrib>
	</ICMSTot>
</total>

```

In test file tags vICMSUFDest and vICMSUFRemet are 0 but in imposto > ICMSUFDest > vICMSUFDest and vICMSUFRemet are 80.00 and 20.00. Should be total of tax 80 and 20 instead of 0?

I've seen in last commit you changed test files. Is there any reason for applied 0 to this values?

Thanks!

**Comment by danimaribeiro** [comment]
@svazquezco In this case the test file is wrong.
You can overwrite it to fix the test.

Those tags ICMSTot are the sum of the lines, because lines have:
```
<vICMSUFDest>80.00</vICMSUFDest>
<vICMSUFRemet>20.00</vICMSUFRemet>
```
The correct in sum is:
```
<vICMSUFDest>80.00</vICMSUFDest>
<vICMSUFRemet>20.00</vICMSUFRemet>
```

**Comment by ghost** [comment]
Thanks @danimaribeiro !

**Comment by ghost** [comment]
@danimaribeiro, Finally I've reverted your changes in last commit for xml file test.

**Pull request merged**